src/Uploadcare/Uploader.php

```diff
@@ -158,29 +158,39 @@ private function fromUrlNew($url, $options = array())
      *
      * @param string $path
      * @param string|bool $mime_type
+     * @param string $filename
      * @return File
      * @throws \Exception
      */
-    public function fromPath($path, $mime_type = false)
+    public function fromPath($path, $mime_type = false, $filename = '')
     {
         if (function_exists('curl_file_create')) {
             if ($mime_type) {
-                $f = curl_file_create($path, $mime_type);
+                $f = curl_file_create($path, $mime_type, $filename);
```

Comment:
this will break of you don't supply filename.

Reply:
I hate this, but what if we could use `call_user_func_array()` here?
```php
$args = array($path, null, null);
if ($mime_type) {
    $args[1] = $mime_type;
}
if ($filename) {
    $args[2] = $filename;
}
call_user_func_array('curl_file_create', $args);
```

Reply:
Made a change that's more compact and does the same as if you pass nulls to call_user_func_array.